certs/service_test.go

```diff
@@ -33,217 +35,300 @@ const (
 	ttl       = "1h"
 	certNum   = 10
 	validID   = "d4ebb847-5d0e-4e46-bdd9-b6aceaaa3a22"
-
-	cfgAuthTimeout = "1s"
-
-	caPath            = "../docker/ssl/certs/ca.crt"
-	caKeyPath         = "../docker/ssl/certs/ca.key"
-	cfgSignHoursValid = "24h"
-	instanceID        = "5de9b29a-feb9-11ed-be56-0242ac120002"
 )
 
-func newService(t *testing.T) (certs.Service, *authmocks.AuthClient, *sdkmocks.SDK) {
+func newService(_ *testing.T) (certs.Service, *mocks.Repository, *mocks.Agent, *authmocks.AuthClient, *sdkmocks.SDK) {
+	repo := new(mocks.Repository)
+	agent := new(mocks.Agent)
 	auth := new(authmocks.AuthClient)
-
 	sdk := new(sdkmocks.SDK)
-	repo := mocks.NewCertsRepository()
-
-	tlsCert, caCert, err := certs.LoadCertificates(caPath, caKeyPath)
-	require.Nil(t, err, fmt.Sprintf("unexpected cert loading error: %s\n", err))
-
-	authTimeout, err := time.ParseDuration(cfgAuthTimeout)
-	require.Nil(t, err, fmt.Sprintf("unexpected auth timeout parsing error: %s\n", err))
 
-	pki := mocks.NewPkiAgent(tlsCert, caCert, cfgSignHoursValid, authTimeout)
+	return certs.New(auth, repo, sdk, agent), repo, agent, auth, sdk
+}
 
-	return certs.New(auth, repo, sdk, pki), auth, sdk
+var cert = certs.Cert{
+	OwnerID: validID,
+	ThingID: thingID,
+	Serial:  "",
+	Expire:  time.Time{},
 }
 
 func TestIssueCert(t *testing.T) {
-	svc, auth, sdk := newService(t)
-
+	svc, repo, agent, auth, sdk := newService(t)
 	cases := []struct {
-		token   string
-		desc    string
-		thingID string
-		ttl     string
-		key     string
-		err     error
+		token        string
+		desc         string
+		thingID      string
+		ttl          string
+		key          string
+		pki          pki.Cert
+		identifyRes  *magistrala.IdentityRes
+		identifyErr  error
+		thingErr     errors.SDKError
+		issueCertErr error
+		repoErr      error
+		err          error
 	}{
 		{
 			desc:    "issue new cert",
 			token:   token,
 			thingID: thingID,
 			ttl:     ttl,
-			err:     nil,
+			pki: pki.Cert{
+				ClientCert:     "",
+				IssuingCA:      "",
+				CAChain:        []string{},
+				ClientKey:      "",
+				PrivateKeyType: "",
+				Serial:         "",
+				Expire:         0,
+			},
+			identifyRes:  &magistrala.IdentityRes{Id: validID},
+			identifyErr:  nil,
+			thingErr:     nil,
+			issueCertErr: nil,
+			err:          nil,
 		},
 		{
 			desc:    "issue new cert for non existing thing id",
 			token:   token,
 			thingID: "2",
 			ttl:     ttl,
-			err:     certs.ErrFailedCertCreation,
+			pki: pki.Cert{
+				ClientCert:     "",
+				IssuingCA:      "",
+				CAChain:        []string{},
+				ClientKey:      "",
+				PrivateKeyType: "",
+				Serial:         "",
+				Expire:         0,
+			},
+			identifyRes:  &magistrala.IdentityRes{Id: validID},
+			identifyErr:  svcerr.ErrAuthentication,
+			thingErr:     errors.NewSDKError(certs.ErrFailedCertCreation),
+			issueCertErr: certs.ErrFailedCertCreation,
+			err:          svcerr.ErrAuthentication,
```

Comment:
In this test case `issue new cert for non existing thing id`
SDK will be returning error 
So only  the filed `thingErr` can be `errors.NewSDKError(certs.ErrFailedCertCreation)`,
But other fields like `identifyErr`, `issueCertErr` and `err` should be nil.